controller/internal/routes/current_identity_authenticator_router.go

```diff
@@ -86,69 +84,19 @@ func (ir *CurrentIdentityAuthenticatorRouter) Detail(ae *env.AppEnv, rc *respons
 }
 
 func (ir *CurrentIdentityAuthenticatorRouter) Update(ae *env.AppEnv, rc *response.RequestContext) {
-	apiEntity := &map[string]interface{}{}
-	Update(rc, ae.Schemes.Authenticator.Put, ir.IdType, apiEntity, func(id string) error {
-		authenticator, err := ae.GetHandlers().Authenticator.ReadForIdentity(rc.Identity, id)
-
-		if err != nil {
-			return err
-		}
-
-		if authenticator == nil {
-			return util.NewNotFoundError(ae.GetHandlers().Authenticator.GetStore().GetSingularEntityType(), "id", id)
-		}
-
-		if authenticator.IdentityId != rc.Identity.Id {
-			return apierror.NewUnhandled()
-		}
-
-		authenticatorUpdate := &AuthenticatorUpdateApi{}
-		authenticatorUpdate.FillFromMap(*apiEntity)
-
-		authenticatorUpdateModel := authenticatorUpdate.ToModel(id)
-
-		if authenticatorUpdateModel.Method != authenticator.Method {
-			return apierror.NewInvalidAuthenticatorProperties()
-		}
-
-		if authenticatorUpdateModel.Method == persistence.MethodAuthenticatorCert {
-			return apierror.NewAuthenticatorCannotBeUpdated()
-		}
-
-		return ae.Handlers.Authenticator.Update(authenticatorUpdateModel)
+	apiEntity := &AuthenticatorSelfUpdateApi{}
+	Update(rc, ae.Schemes.AuthenticatorSelf.Put, ir.IdType, apiEntity, func(id string) error {
+		return ae.Handlers.Authenticator.UpdateSelf(apiEntity.ToModel(id, rc.Identity.Id))
 	})
 }
 
 func (ir *CurrentIdentityAuthenticatorRouter) Patch(ae *env.AppEnv, rc *response.RequestContext) {
-	apiEntity := &map[string]interface{}{}
-	Patch(rc, ae.Schemes.Authenticator.Patch, ir.IdType, apiEntity, func(id string, fields JsonFields) error {
-		authenticator, err := ae.GetHandlers().Authenticator.ReadForIdentity(rc.Identity, id)
-
-		if err != nil {
-			return err
-		}
-
-		if authenticator == nil {
-			return util.NewNotFoundError(ae.GetHandlers().Authenticator.GetStore().GetSingularEntityType(), "id", id)
-		}
-
-		if authenticator.IdentityId != rc.Identity.Id {
-			return apierror.NewUnhandled()
+	apiEntity := &AuthenticatorSelfUpdateApi{}
+	Patch(rc, ae.Schemes.AuthenticatorSelf.Patch, ir.IdType, apiEntity, func(id string, fields JsonFields) error {
+		if fields.IsUpdated("newPassword") {
```

Comment:
@plorenz there I had to add this as there is a computed field (salt) and fields in the API that don't match the model/persistence field names.

I don't know if there is a better way, but this was "easy." I wanted to do it in the handlers (so that any presentation would have the logic), but I didn't know if extending the boltz checker interface was the right call as it could be a conjunction of the handler and json fields.

Reply:
There's an AndFieldChecker which combines two checkers. You could do a similar thing with an OrFieldChecker, which would be pretty simple.

However, this isn't offensive either :)

Reply:
I want it in the handler - any logic outside the handlers means it has to be duplicated for other presentations.

I've added the disjunction field checker.